src/runtime/hexagon/hexagon_hvx.cc

```diff
@@ -31,25 +31,30 @@ namespace tvm {
 namespace runtime {
 namespace hexagon {
 
-HexagonHvx::HexagonHvx() {
-  // Reserve HVX.
-  int res = qurt_hvx_reserve(QURT_HVX_RESERVE_ALL_AVAILABLE);
-  CHECK((res != QURT_HVX_RESERVE_NOT_SUPPORTED) && (res != QURT_HVX_RESERVE_NOT_SUCCESSFUL))
-      << "error reserving HVX: " << res;
+HexagonHvx::HexagonHvx() { Acquire(); }
 
-  // Lock HVX.
+HexagonHvx::~HexagonHvx() { Release(); }
+
+void HexagonHvx::Acquire() {
+  reserved_count_ = qurt_hvx_reserve(QURT_HVX_RESERVE_ALL);
+  CHECK((reserved_count_ == QURT_HVX_RESERVE_ALL) ||
+        (reserved_count_ == QURT_HVX_RESERVE_ALREADY_MADE))
```

Comment:
I also considered making it a failure if another instance of HexagonHvx

Reply:
Yeah I agree that this should probably be an error if `reserved_count_ == QURT_HVX_RESERVE_ALREADY_MADE`.

Reply:
Will push an update - thanks!